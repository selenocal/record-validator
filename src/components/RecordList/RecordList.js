import React from "react";
import styled from "styled-components";
import PropTypes from "prop-types";

const RecordList = ({ records }) => {
  return (
    <Wrapper>
      <TableContainer>
        {records.length > 0 ? (
          <Table>
            <thead>
              <tr>
                <th>Reference</th>
                <th>Description</th>
              </tr>
            </thead>
            <tbody>
              {records.map(record => (
                <tr
                  key={
                    record.attributes
                      ? record.attributes.reference + record.children[0].value
                      : record[0] + record[1]
                  }
                >
                  <td>
                    {record.attributes
                      ? record.attributes.reference
                      : record[0]}
                  </td>
                  <td>
                    {record.children ? record.children[1].value : record[2]}
                  </td>
                </tr>
              ))}
            </tbody>
          </Table>
        ) : (
          <WarningText>No results</WarningText>
        )}
      </TableContainer>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  width: 100%;
  border: 1px solid #c7c8c8;
  background: white;
`;

const TableContainer = styled.div`
  padding: 16px;
`;

const Table = styled.table`
  width: 100%;
  thead {
    background-color: #f8f8f8;
  }
  th {
    border: 1px solid #ececec;
    text-transform: capitalize;
    padding: 5px 15px;
    font-size: 16px;
  }
  td {
    text-align: center;
    vertical-align: middle;
    padding: 15px 15px 10px;
    border-bottom: 1px solid #ececec;
    color: #1f2223;
    font-size: 16px;
  }
`;

const WarningText = styled.div`
  text-align: center;
`;

RecordList.propTypes = {
  records: PropTypes.array
};

RecordList.defaultProps = {
  records: []
};
export default RecordList;
