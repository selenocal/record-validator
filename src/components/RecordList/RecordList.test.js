import React from "react";
import { shallow } from "enzyme";
import RecordList from "./RecordList";

describe("<RecordList />", () => {
  const fileContent = [
    {
      name: "record",
      attributes: {
        reference: "189177"
      },
      children: [
        {
          name: "accountNumber",
          value: "NL27SNSB0917829871"
        },
        {
          name: "description",
          value: "Subscription for Erik Dekker"
        },
        {
          name: "startBalance",
          value: "5429"
        },
        {
          name: "mutation",
          value: "-939"
        },
        {
          name: "endBalance",
          value: "6368"
        }
      ]
    },
    {
      name: "record",
      attributes: {
        reference: "189177"
      },
      children: [
        {
          name: "accountNumber",
          value: "NL27SNSB0917829871"
        },
        {
          name: "description",
          value: "Subscription for Erik Dekker"
        },
        {
          name: "startBalance",
          value: "5429"
        },
        {
          name: "mutation",
          value: "-939"
        },
        {
          name: "endBalance",
          value: "6368"
        }
      ]
    }
  ];

  it("expects render empty RecordList correctly", () => {
    const wrapper = shallow(<RecordList />);
    expect(wrapper).toMatchSnapshot();
  });

  it("expects render RecordList with mock data correctly", () => {
    const wrapper = shallow(<RecordList records={fileContent} />);
    expect(wrapper).toMatchSnapshot();
    fileContent.forEach(record => {
      expect(wrapper.contains(record.attributes.reference)).toBe(true);
      expect(wrapper.contains(record.children[1].value)).toBe(true);
    });
  });
});
