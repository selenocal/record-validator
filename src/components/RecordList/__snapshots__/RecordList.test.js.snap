// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RecordList /> expects render RecordList with mock data correctly 1`] = `ShallowWrapper {}`;

exports[`<RecordList /> expects render empty RecordList correctly 1`] = `ShallowWrapper {}`;
